locallib.php

```diff
@@ -235,11 +239,13 @@ function subcourse_grades_update($courseid, $subcourseid, $refcourseid, $itemnam
 
         $gs = grade_grade::fetch_all(['itemid' => $gi->id]);
 
-        foreach ($gs as $g) {
-            if (isset($refgrades->grades[$g->userid])) {
-                if ($refgrades->grades[$g->userid]->hidden != $g->hidden) {
-                    $g->grade_item = $gi;
-                    $g->set_hidden($refgrades->grades[$g->userid]->hidden);
+        if ($gs) {
```

Comment:
Good catch on this, I missed the `Array of object instances or false if not found` docs